src/app/store/effects/VehicleTestResultModel.effects.ts

```diff
@@ -1,32 +1,73 @@
-import {Injectable} from '@angular/core';
+import { Injectable } from '@angular/core';
 import {
-  EVehicleTestResultModelActions, GetVehicleTestResultModel, GetVehicleTestResultModelSuccess
+  EVehicleTestResultModelActions,
+  GetVehicleTestResultModel,
+  GetVehicleTestResultModelSuccess,
+  SetCurrentState,
+  UpdateTestResult,
+  UpdateTestResultSuccess
 } from '@app/store/actions/VehicleTestResultModel.actions';
-import {TestResultService} from '@app/technical-record-search/test-result.service';
-import {Actions, Effect, ofType} from '@ngrx/effects';
-import {Action, Store} from '@ngrx/store';
-import {Observable, of} from 'rxjs';
-import {map, switchMap} from 'rxjs/operators';
-import {ClearErrorMessage} from '../actions/Error.actions';
-import {IAppState} from '../state/app.state';
+import { TestResultService } from '@app/technical-record-search/test-result.service';
+import { Actions, Effect, ofType } from '@ngrx/effects';
+import { Action, Store } from '@ngrx/store';
+import { Observable, of } from 'rxjs';
+import { catchError, map, switchMap } from 'rxjs/operators';
+import { ClearErrorMessage, SetErrorMessage } from '../actions/Error.actions';
+import { IAppState } from '../state/app.state';
+import { TestResultModel } from '@app/models/test-result.model';
+import { VehicleTestResultUpdate } from '@app/models/vehicle-test-result-update';
+import { VIEW_STATE } from '@app/app.enums';
+import {TestResultTestTypeNumber} from '@app/models/test-result-test-type-number';
+import {UserService} from '@app/app-user.service';
 
 @Injectable()
 export class VehicleTestResultModelEffects {
   @Effect()
   getTestResults$: Observable<Action> = this._actions$.pipe(
     ofType<GetVehicleTestResultModel>(EVehicleTestResultModelActions.GetVehicleTestResultModel),
-    map(action => action.payload),
-    switchMap((searchIdentifier: string) => this._testResultService.getTestResults(searchIdentifier)),
-    switchMap((testResultJson: any) => {
-      this._store$.dispatch(new ClearErrorMessage());
-      return of(new GetVehicleTestResultModelSuccess(testResultJson));
+    map((action) => action.payload),
+    switchMap((searchIdentifier: string) => {
+      return this._testResultService.getTestResults(searchIdentifier);
     }),
+    switchMap((testResults: TestResultModel[]) => {
+      this._store.dispatch(new ClearErrorMessage());
+      return of(new GetVehicleTestResultModelSuccess(testResults));
+    })
+  );
+
+  @Effect()
+  updateTestResult$ = this._actions$.pipe(
+    ofType<UpdateTestResult>(EVehicleTestResultModelActions.UpdateTestResult),
+    map((action) => action.testResultTestTypeNumber),
+    switchMap((testResultTestTypeNumber: TestResultTestTypeNumber) => {
+
+      const testResultTestTypeNumberObject = testResultTestTypeNumber;
```

Comment:
It seems that we need certain props from `testResultTestTypeNumber ` here to reconstruct a new object
Can we do something like to make it simple instead rather than creating this temporary object?
```
// pulling out testResultUpdated from testResultTestTypeNumber
const { testResultUpdated } = testResultTestTypeNumber
...
testResultObject.testResult = testResultUpdated
```